enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -1758,13 +1758,17 @@ func (p *PebbleCache) deleteMetadataOnly(ctx context.Context, key filestore.Pebb
 	}
 	defer db.Close()
 
+	unlockFn := p.locker.RLock(key.LockID())
```

Comment:
deleteMetadataOnly is called from handleMetadataMismatch and both places that call handleMetadataMismatch have the lock already.

Reply:
You could add a comment to the function mentioning that a key lock should already be held before calling the function.